lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
```

Comment:
Fixed and a slightly more simplified version (could be even better if Node would already support rest parameters):

```javascript
'use strict';

module.exports = promisify;

function promisify(fn, single) {
  Object.defineProperty(inner, 'name', {
    value: `${fn.name}Promisified`
  });
  return inner;

  function inner() {
    return new Promise((resolve, reject) => {
      const args = Array.from(arguments);
      args.push(single ? resolve : function(err, value) {
        if (err) reject(err);
        resolve(args.length <= 2 ? value : args.slice(1));
      });
      fn.apply(this, arguments);
    });
  }
}
```